<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="3" ignored="0" total="7" passed="4">
  <reporter-output>
  </reporter-output>
  <suite name="用户管理系统测试套件" duration-ms="11654" started-at="2020-09-20T05:03:14Z" finished-at="2020-09-20T05:03:26Z">
    <groups>
      <group name="loginTrue">
        <method signature="LoginTest.loginTrue()[pri:0, instance:com.course.cases.LoginTest@34ce8af7]" name="loginTrue" class="com.course.cases.LoginTest"/>
      </group> <!-- loginTrue -->
    </groups>
    <test name="用户管理系统测试用例" duration-ms="11654" started-at="2020-09-20T05:03:14Z" finished-at="2020-09-20T05:03:26Z">
      <class name="com.course.cases.UpdateUserInfoTest">
        <test-method status="FAIL" signature="updateUserInfo()[pri:0, instance:com.course.cases.UpdateUserInfoTest@7f63425a]" name="updateUserInfo" duration-ms="2061" depends-on-groups="loginTrue" started-at="2020-09-20T13:03:19Z" description="更改用户信息" finished-at="2020-09-20T13:03:21Z">
          <exception class="org.apache.ibatis.exceptions.PersistenceException">
            <message>
              <![CDATA[
### Error querying database.  Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
### Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.apache.ibatis.exceptions.PersistenceException: 
### Error querying database.  Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
### Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
	at org.apache.ibatis.exceptions.ExceptionFactory.wrapException(ExceptionFactory.java:30)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:150)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:141)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectOne(DefaultSqlSession.java:77)
	at com.course.cases.UpdateUserInfoTest.updateUserInfo(UpdateUserInfoTest.java:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:74)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:93)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:101)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.booter.ProviderFactory$ClassLoaderProxy.invoke(ProviderFactory.java:103)
	at com.sun.proxy.$Proxy0.invoke(Unknown Source)
	at org.apache.maven.surefire.booter.SurefireStarter.invokeProvider(SurefireStarter.java:145)
	at org.apache.maven.surefire.booter.SurefireStarter.runSuitesInProcess(SurefireStarter.java:87)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:69)
Caused by: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
	at org.apache.ibatis.session.Configuration$StrictMap.get(Configuration.java:875)
	at org.apache.ibatis.session.Configuration.getMappedStatement(Configuration.java:708)
	at org.apache.ibatis.session.Configuration.getMappedStatement(Configuration.java:701)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:147)
	... 37 more
]]>
            </full-stacktrace>
          </exception> <!-- org.apache.ibatis.exceptions.PersistenceException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- updateUserInfo -->
        <test-method status="FAIL" signature="deleteUser()[pri:0, instance:com.course.cases.UpdateUserInfoTest@7f63425a]" name="deleteUser" duration-ms="3050" depends-on-groups="loginTrue" started-at="2020-09-20T13:03:21Z" description="删除用户" finished-at="2020-09-20T13:03:24Z">
          <exception class="org.apache.ibatis.exceptions.PersistenceException">
            <message>
              <![CDATA[
### Error querying database.  Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
### Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.apache.ibatis.exceptions.PersistenceException: 
### Error querying database.  Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
### Cause: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
	at org.apache.ibatis.exceptions.ExceptionFactory.wrapException(ExceptionFactory.java:30)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:150)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:141)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectOne(DefaultSqlSession.java:77)
	at com.course.cases.UpdateUserInfoTest.deleteUser(UpdateUserInfoTest.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:74)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:93)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:101)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.booter.ProviderFactory$ClassLoaderProxy.invoke(ProviderFactory.java:103)
	at com.sun.proxy.$Proxy0.invoke(Unknown Source)
	at org.apache.maven.surefire.booter.SurefireStarter.invokeProvider(SurefireStarter.java:145)
	at org.apache.maven.surefire.booter.SurefireStarter.runSuitesInProcess(SurefireStarter.java:87)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:69)
Caused by: java.lang.IllegalArgumentException: Mapped Statements collection does not contain value for null
	at org.apache.ibatis.session.Configuration$StrictMap.get(Configuration.java:875)
	at org.apache.ibatis.session.Configuration.getMappedStatement(Configuration.java:708)
	at org.apache.ibatis.session.Configuration.getMappedStatement(Configuration.java:701)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:147)
	... 37 more
]]>
            </full-stacktrace>
          </exception> <!-- org.apache.ibatis.exceptions.PersistenceException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- deleteUser -->
      </class> <!-- com.course.cases.UpdateUserInfoTest -->
      <class name="com.course.cases.LoginTest">
        <test-method status="PASS" signature="beforTest()[pri:0, instance:com.course.cases.LoginTest@34ce8af7]" name="beforTest" is-config="true" duration-ms="88" started-at="2020-09-20T13:03:14Z" description="测试准备工作，获取HttpClient对象" finished-at="2020-09-20T13:03:14Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforTest -->
        <test-method status="PASS" signature="loginTrue()[pri:0, instance:com.course.cases.LoginTest@34ce8af7]" name="loginTrue" duration-ms="2121" started-at="2020-09-20T13:03:14Z" description="用户成功登陆接口" finished-at="2020-09-20T13:03:17Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTrue -->
        <test-method status="PASS" signature="loginFalse()[pri:0, instance:com.course.cases.LoginTest@34ce8af7]" name="loginFalse" duration-ms="88" started-at="2020-09-20T13:03:17Z" description="用户登陆失败接口" finished-at="2020-09-20T13:03:17Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginFalse -->
      </class> <!-- com.course.cases.LoginTest -->
      <class name="com.course.cases.GetUserInfoTest">
        <test-method status="PASS" signature="getUserInfo()[pri:0, instance:com.course.cases.GetUserInfoTest@3f3afe78]" name="getUserInfo" duration-ms="2069" depends-on-groups="loginTrue" started-at="2020-09-20T13:03:17Z" description="获取userId为1的用户信息" finished-at="2020-09-20T13:03:19Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getUserInfo -->
      </class> <!-- com.course.cases.GetUserInfoTest -->
      <class name="com.course.cases.AddUserTest">
        <test-method status="FAIL" signature="addUser()[pri:0, instance:com.course.cases.AddUserTest@b684286]" name="addUser" duration-ms="71" depends-on-groups="loginTrue" started-at="2020-09-20T13:03:17Z" description="添加用户接口接口" finished-at="2020-09-20T13:03:17Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at com.course.cases.AddUserTest.addUser(AddUserTest.java:22)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:74)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:93)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:101)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.booter.ProviderFactory$ClassLoaderProxy.invoke(ProviderFactory.java:103)
	at com.sun.proxy.$Proxy0.invoke(Unknown Source)
	at org.apache.maven.surefire.booter.SurefireStarter.invokeProvider(SurefireStarter.java:145)
	at org.apache.maven.surefire.booter.SurefireStarter.runSuitesInProcess(SurefireStarter.java:87)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:69)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addUser -->
      </class> <!-- com.course.cases.AddUserTest -->
      <class name="com.course.cases.GetUserInfoListTest">
        <test-method status="PASS" signature="getUserListInfo()[pri:0, instance:com.course.cases.GetUserInfoListTest@36d64342]" name="getUserListInfo" duration-ms="2077" depends-on-groups="loginTrue" started-at="2020-09-20T13:03:24Z" description="获取性别为男的用户信息" finished-at="2020-09-20T13:03:26Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getUserListInfo -->
      </class> <!-- com.course.cases.GetUserInfoListTest -->
    </test> <!-- 用户管理系统测试用例 -->
  </suite> <!-- 用户管理系统测试套件 -->
</testng-results>
